coffee/shepherd.coffee

```diff
@@ -117,6 +117,9 @@ class Step extends Evented
     @tether = new Tether extend(tetherOpts, @options.tetherOptions)
 
   show: =>
+      @options.beforeShowPromise?()?.then(=> @_show()) ? @_show()
```

Comment:
@b-ash They are not the same. They do not yield the same JS. Changing it to ```.then(@_show)``` removes the anonymous function from around it.

```js
Step.prototype.show = function() {
      var _base, _ref1, _ref2;
      return (_ref1 = typeof (_base = this.options).beforeShowPromise === "function" ? (_ref2 = _base.beforeShowPromise()) != null ? _ref2.then(this._show) : void 0 : void 0) != null ? _ref1 : this._show();
    };
```

vs this, where it's wrapped in ```function```
```js
Step.prototype.show = function() {
      var _base, _ref1, _ref2,
        _this = this;
      return (_ref1 = typeof (_base = this.options).beforeShowPromise === "function" ? (_ref2 = _base.beforeShowPromise()) != null ? _ref2.then(function() {
        return _this._show();
      }) : void 0 : void 0) != null ? _ref1 : this._show();
    };
```

The code does not work, if I switch it to the way you suggested.